docs/src/main/asciidoc/se/guides/upgrade_4x.adoc

```diff
@@ -122,6 +122,64 @@ Features are not registered immediately—each feature can define a `Weight` or
 This is to allow ordering of features in a meaningful way (e.g. Context should be first, Tracing second, Security third etc).
 
 
+== Adding Additional Routing Criteria
+
+In prior versions, the `RequestPredicate` class was used to specify more routing criteria. This class has been removed from Helidon 4. 
+
+So, for example, if you used the following in Helidon 3:
+
+[source, java]
+.Helidon 3.x using `RequestPredicate` 
```

Comment:
These snippets should go in `src/main/java/io/helidon/docs/se/guides/Upgrade4xSnippets.java`.
The 3.x snippet cannot be compiled and should be commented-out, you can see how it's done for the current snippets in that class.